woocommerce/woocommerce-gateway-stripe
# Subscription renewal orders are missing the statement descriptor in Stripe

Fixes #3450

## Changes proposed in this Pull Request:







This PR addresses the issue where subscription renewal orders are missing the `statement_descriptor_suffix` parameter, which is necessary after the deprecation of the `statement_descriptor` parameter in January 2024.

| Parent | Renewal :x: |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/4fe51bbd-07a7-45ae-804f-f3a54649df1c)|![image](https://github.com/user-attachments/assets/606f306f-941b-4833-8a31-aa6f11cc9fd0)| 

In https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2833 we updated all of our code to use `statement_descriptor_suffix` instead of the deprecated `statement_descriptor` but I believe the `create_and_confirm_intent_for_off_session` function was missed in the PR, which impacts renewal orders.

## Testing instructions





1. In your stripe settings, enable the "Add customer order number to the bank statement" setting
2. Purchase a subscription product with Stripe
3. Go to https://dashboard.stripe.com/test/payments and view the parent order and confirm the order number in statement descriptor is present:
![image](https://github.com/user-attachments/assets/07ca5140-deed-4169-adcb-7d00c8e80b00)
4. Process a renewal order and view the latest charge in stripe and confirm the `statement_descriptor_suffix` param is now populated while on this branch

---


**Post merge**
